### PR TITLE
17 no start tag regex

### LINKED ISSUE #17: Prevent recursion error
The following template crashes the browser:

```
<ul id="todo-list">
  {{#each todos}}
    <li class="todo {{#if complete}}completed{{/if}}
      {{#if isDestroying}}destroying{{/if}}
      {{#if isEditing(this)}}editing{{/if}}">
      <div class="view">
        <input class="toggle" type="checkbox" {($checked)}="complete" ($change)=>
        <label ($dblclick)="edit(this)">{{name}}</label>
        <button class="destroy" ($click)="destroy()"></button>
      </div>
      <input class="edit" type="text"
        {($value)}="name"
        ($enter)="updateName()"
        {$focused}="isEditing(this)"
        ($blur)="cancelEdit()"/>
    </li>
  {{/each}}
</ul>
```

### PR DESCRIPTION
**Pull request by iherndon**
fixes #17

### CODE REVIEW

File: can-view-parser.js [hunk 1]
```diff
@@ -194,14 +179,15 @@ var HTMLParser = function (html, handler, returnIntermediate) {
 
 				// start tag
 			} else if (html.indexOf("<") === 0) {
-				match = html.match(startTag);
+				var res = HTMLParser.handleStartTag(html);
```
Comment (justinbmeyer) [inline can-view-parser.js]: maybe instead of `handleStartTag` maybe it should be `searchStartTag`?

File: can-view-parser.js [hunk 2]
```diff
@@ -365,6 +351,9 @@ HTMLParser.parseAttrs = function(rest, handler){
 			// if we haven't yet started this attribute `{{}}=foo` case:
 			if(!state.attrStart) {
 				callAttrStart(state, curIndex, handler, rest);
+				if(i === rest.length){
```
Comment (justinbmeyer) [inline can-view-parser.js]: Why was this added?

File: can-view-parser.js (hunk 1 above)
Comment (iherndon) [inline can-view-parser.js]: I will update this.

File: can-view-parser.js (hunk 2 above)
Comment (iherndon) [inline can-view-parser.js]: I will add a comment

**Pull request merged**